src/cljc/markdown/transformers.cljc

```diff
@@ -125,9 +125,9 @@
     (str " " text) text))
 
 (defn paragraph
-  [text {:keys [eof heading inline-heading temp hr code lists blockquote paragraph last-line-empty?] :as state}]
+  [text {:keys [eof heading inline-heading temp hr code codeblock lists blockquote paragraph last-line-empty? end-codeblock?] :as state}]
   (cond
-    (and paragraph lists)
+    (or (and paragraph lists) (and paragraph codeblock))
```

Comment:
However codeblocks are not actually mutually exclusive from paragraphs. as seen here:
`user=> (md-to-html-string "hey\n```\nhey\n```\nhey")
"<p>hey<pre><code>hey\n </code></pre> hey</p>"`
Note the paragraph is ending after the codeblock. This is what is causing the codeblock to take up an extra line. The above code says *if you are in a paragraph while in a codeblock, end the paragraph.* The rest of the code solves the issue of making sure the paragraph starts back up again after the codeblock.